tests/mod_api/test_controller.py

```diff
@@ -898,3 +898,167 @@ def tearDown(self):
+        self.assertEqual(response.status_code, 200)
+
+    @patch('app.mod_api.models.BitStore.copy_to_new_version')
+    @patch('app.mod_api.models.MetaDataDB.create_or_update_version')
+    def test_throw_400_if_version_missing(self, create_or_update_version, copy_to_new_version):
+        copy_to_new_version.return_value = True
+        create_or_update_version.return_value = True
+        response = self.client.post(self.url,
+                                    data=json.dumps({
+                                        'version_fail': 'tag_one'
+                                    }),
+                                    content_type='application/json',
+                                    headers=dict(Authorization=self.auth))
+        self.assertEqual(response.status_code, 400)
+        data = json.loads(response.data)
+        self.assertEqual('ATTRIBUTE_MISSING', data['error_code'])
+
+    @patch('app.mod_api.models.BitStore.copy_to_new_version')
+    @patch('app.mod_api.models.MetaDataDB.create_or_update_version')
+    def test_throw_500_if_failed_to_tag(self, create_or_update_version, copy_to_new_version):
+        copy_to_new_version.return_value = False
+        create_or_update_version.return_value = True
+        response = self.client.post(self.url,
+                                    data=json.dumps({
+                                        'version': 'tag_one'
+                                    }),
+                                    content_type='application/json',
+                                    headers=dict(Authorization=self.auth))
+        self.assertEqual(response.status_code, 500)
+
+    @patch('app.mod_api.models.BitStore.copy_to_new_version')
+    @patch('app.mod_api.models.MetaDataDB.create_or_update_version')
+    def test_throw_403_if_not_owner_or_member_of_publisher(self, create_or_update_version,
+                                                           copy_to_new_version):
+        copy_to_new_version.return_value = False
+        create_or_update_version.return_value = True
+        response = self.client.post(self.jwt_url,
+                                    data=json.dumps({
+                                        'username': self.user_not_allowed_name,
+                                        'secret': 'super_secret'
+                                    }),
+                                    content_type='application/json')
+        data = json.loads(response.data)
+        jwt_not_allowed = data['token']
+        auth_not_allowed = "bearer %s" % jwt_not_allowed
+
+        response = self.client.post(self.url,
+                                    data=json.dumps({
+                                        'version': 'tag_one'
+                                    }),
+                                    content_type='application/json',
+                                    headers=dict(Authorization=auth_not_allowed))
+        self.assertEqual(response.status_code, 403)
```

Comment:
Well, you can just check those two mocks calls, but there may be other places modifying database somwhere. So i would recommend inspecting the actual database contents and calls to boto3 for the sake of more strict security checks.